scrapinghub/client.py

```diff
@@ -1051,6 +1051,164 @@ def post(self, _value, **kwargs):
+        self._proxy_methods(['close', 'flush'])
+
+    def get(self, name):
+        """Get a frontier by name."""
+        return Frontier(self._client, self, name)
+
+    def iter(self):
+        """Iterate through frontiers."""
+        return iter(self.list())
+
+    def list(self):
+        """List frontiers."""
+        return next(self._origin.apiget('list'))
+
+    @property
+    def newcount(self):
+        return self._origin.newcount
+
+
+class Frontier(object):
+    """Representation of a frontier object.
+
+    Not a public constructor: use :class:`Frontiers` instance to get a
+    :class:`Frontier` instance. See :meth:`Frontiers.get` method.
+
+    Usage:
+
+    - get iterator with all slots
+        >>> frontier.iter()
+        <list_iterator at 0x1030736d8>
+
+    - list all slots
+        >>> frontier.list()
+        ['example.com', 'example.com2']
+
+    - get a slot by name
+        >>> frontier.get('example.com')
+        <scrapinghub.client.FrontierSlot at 0x1049d8978>
+
+    - flush frontier data
+        >>> frontier.flush()
+    """
+    def __init__(self, client, frontiers, name):
+        self.key = name
+        self._client = client
+        self._frontiers = frontiers
+
+    def get(self, slot):
+        """Get a slot by name."""
+        return FrontierSlot(self._client, self, slot)
+
+    def iter(self):
+        """Iterate through slots."""
+        return iter(self.list())
+
+    def list(self):
+        """List all slots."""
+        return next(self._frontiers._origin.apiget((self.key, 'list')))
+
+    def flush(self):
+        """Flush data for a whole frontier."""
+        writers = self._frontiers._origin._writers
+        for (fname, _), writer in writers.items():
+            if fname == self.key:
+                writer.flush()
+
+
+class FrontierSlot(object):
+    """Representation of a frontier slot object.
+
+    Not a public constructor: use :class:`Frontier` instance to get a
+    :class:`FrontierSlot` instance. See :meth:`Frontier.get` method.
+
+    Usage:
+
+    - add request to a slot
+        >>> data = [{'fp': 'page1.html', 'p': 1, 'qdata': {'depth': 1}}]
+        >>> slot.add('example.com', data)
+
+    - flush data for a slot
+        >>> slot.flush()
+
+    - read requests from a slot
+        >>> slot.iter()
+        <generator object jldecode at 0x1049aa9e8>
+        >>> slot.list()
+        [{'id': '0115a8579633600006',
+          'requests': [['page1.html', {'depth': 1}]]}]
+
+    - delete a batch with requests from a slot
+        >>> slot.delete('0115a8579633600006')
+
+    - delete a whole slot
+        >>> slot.delete()
+
+    """
+    def __init__(self, client, frontier, slot):
+        self.key = slot
+        self._client = client
+        self._frontier = frontier
+
+    def add(self, fps):
+        """Add requests to slot."""
+        origin = self._frontier._frontiers._origin
+        return origin.add(self._frontier.key, self.key, fps)
+
+    def iter(self, mincount=None):
```

Comment:
better replace `mincount` with `**kwargs`, it's server responsibility to decide what parameters are valid or not